drivers/gpio/hcsr04_driver.go

```diff
@@ -0,0 +1,183 @@
+package hcsr04
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"gobot.io/x/gobot"
+	"gobot.io/x/gobot/drivers/gpio"
+)
+
+const (
+	soundSpeed       float32       = 343.0
+	measurementCycle time.Duration = 60 // 60ms between two measurements
+	// MonitorUpdate is the time between each monitor update
+	MonitorUpdate time.Duration = 100 * time.Millisecond
+)
+
+// HCSR04 instance
+type HCSR04 struct {
+	name       string
+	connection gobot.Adaptor
+	triggerPin *gpio.DirectPinDriver
+	echoPin    *gpio.DirectPinDriver
+	// triggerPin             rpio.Pin
+	// echoPin                rpio.Pin
+	mux                    sync.Mutex
+	Measure                float32 // The last measure
+	distanceMonitorControl chan int
+	distanceMonitorStarted bool
+	gobot.Commander
+}
+
+// NewHCSR04 creates a new HCSR04 instance
+func NewHCSR04(a gobot.Adaptor, triggerPinID string, echoPinID string) *HCSR04 {
+	hcsr04 := &HCSR04{
+		name:       gobot.DefaultName("HCSR04"),
+		triggerPin: gpio.NewDirectPinDriver(a, triggerPinID),
+		echoPin:    gpio.NewDirectPinDriver(a, echoPinID),
+		connection: a,
+		Commander:  gobot.NewCommander(),
+	}
+
+	// hcsr04 := HCSR04{
+	// 	triggerPinID: triggerPinID,
+	// 	echoPinID:    echoPinID,
+	// }
+	// hcsr04.triggerPin = rpio.Pin(hcsr04.triggerPinID)
+	// hcsr04.triggerPin.Mode(rpio.Output)
+
+	// hcsr04.echoPin = rpio.Pin(hcsr04.echoPinID)
+	// hcsr04.echoPin.Mode(rpio.Input)
+	// hcsr04.echoPin.PullDown()
```

Comment:
the echo pin possibly needs a pull down resistor, do you have add one?

Reply:
Yes, I put one